frontend/src/store/store.jsx

```diff
@@ -0,0 +1,46 @@
+import React, { createContext, useReducer, useEffect } from 'react';
+
+// Default state for the store
+const initialStore = {
+  sessionID: '', // Empty string represents user not authenticated, otherwise they are authenticated
+  plan: '', // "FREE" or "ENTERPRISE"
+};
+
+// On app load, try to fetch a saved store from localStorage
+const localStore = JSON.parse(localStorage.getItem('store'));
+
+// Updates the store with newStoreValues
+// Pass `newStoreValues = null` to reset the store
+const storeReducer = (store, newStoreValues) => {
```

Comment:
I like your solution with reducer but I think it could be simplified a bit (at least for this app). For example, we can do something like this:

```tsx
function AppContext (props) {
  const [ state, setState ] = React.useState(() => {
    const session = localStorage.getItem('session');
    return {
      session
    }    
  })

  const setSession = session = {
    localStorage.setItem('session', JSON.stringify(session);
    setState({
      ...state,
      session
    })
  }

  const clearSession = () => {
    setSession(null)    
  }
  
  const ctx = {
    ...state,
    setSession,
    clearSession
  }

  return (
    <StoreContext.Provider value={ctx} children={props.children}/>      
  );
}

```
Also, you may want to create a file `session.ts` with 2 helper methods, so you can get the token outside from virtual DOM (like your api.ts).

```ts
export function setSession(session){
  localStorage.setItem('session', JSON.stringify(session);
}

export function getSession(){
  return JSON.parse(localStorage.getItem('session'))
}
```

But again, your reducer looks good enough to me.

Reply:
Makes sense, agreed that reducer is overkill for the scope of this app.

I've had the token issue in the back of my mind but hadn't come up with the best way to solve it. Your
```javascript
export function getSession(){
  return JSON.parse(localStorage.getItem('session'))
}
```
suggestion is the simplest to implement and should work to allow the `api` to grab the token. The main concern here would be a possible performance hit in the future caused by needing to `JSON.parse` the session object on every call. A better integrated design could be making the `api` part of the `AppContext` to give it direct access to the global state. But as long as you're ok with the quicker-to-implement `getSession` solution for this project I'll go with that.

Unless I'm missing something, I don't think I'll need `setSession`, I can just use `AppContext.setSession` everywhere.

Reply:
> Unless I'm missing something, I don't think I'll need setSession

This is to keep localStorage getters/setters method in one place for tracking purposes (like item key names, etc). Then you can call it from within AppContext.setSession (instead of referencing the localStorage directly).

Reply:
> But as long as you're ok with the quicker-to-implement getSession solution for this project I'll go with that.

Yeah, i am fine with this performance overhead in favor of a better decoupling between components and ease of use. Also, I would not worry about performance optimizations at such level at this stage.

Reply:
> > Unless I'm missing something, I don't think I'll need setSession
> 
> This is to keep localStorage getters/setters methods in one place for tracking purposes (like item key names, etc). Then you can call it from within AppContext.setSession (instead of referencing the localStorage directly).

Ah I see, I was confused but now I get you



> > But as long as you're ok with the quicker-to-implement getSession solution for this project I'll go with that.
> 
> Yeah, i am fine with this performance overhead in favor of a better decoupling between components and ease of use. Also, I would not worry about performance optimizations at such level at this stage.

Roger that

Reply:
> Also, you may want to create a file session.ts 

(correction) instead of creating `sesion.ts`, you can create `localStorage.ts` and put these methods in there. So all window.localStorage setter/getters are in 1 place.